src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -3544,6 +3544,442 @@ class StubGenerator: public StubCodeGenerator {
+        // non_match is not referenced again.  Only CCR6-EQ matters.
+        __ vcmpequb_(non_match, non_match, vec_0s);
+        __ bne_predict_not_taken(CCR6, unrolled_loop_exit);
+
+        // The Base64 characters had no errors, so add the offsets
+        __ vaddubm(input, input, offsets);
+
+        // Pack
+        //
+        // Legend for the tables below: b0, b1, .. b15 are the bytes of
+        // decoded binary data.  The specifier after the colon depicts
+        // which bits are there.  The bit numbering is big endian style
+        // (bit 0 is the most significant).  The || is a concatenate
+        // operator (same terminology as used in the Power ISA 3.x
+        // document).  Strings of 0's are a field of zeros with the shown
+        // length.
+
+        if (PowerArchitecturePPC64 >= 10) {
+            // Note that only e15..e8 are shown here because the extract
+            // bit pattern is the same in e7..e0.
+            //
+            // +===============+=============+======================+======================+=============+=============+======================+======================+=============+
+            // |    Vector     |     e15     |         e14          |         e13          |     e12     |     e11     |         e10          |          e9          |     e8      |
+            // |    Element    |             |                      |                      |             |             |                      |                      |             |
+            // +===============+=============+======================+======================+=============+=============+======================+======================+=============+
+            // | after vaddubm | 00||b0:0..5 | 00||b0:6..7||b1:0..3 | 00||b1:4..7||b2:0..1 | 00||b2:2..7 | 00||b3:0..5 | 00||b3:6..7||b4:0..3 | 00||b4:4..7||b5:0..1 | 00||b5:2..7 |
+            // +---------------+-------------+----------------------+----------------------+-------------+-------------+----------------------+----------------------+-------------+
+            // |  after xxbrd  | 00||b5:2..7 | 00||b4:4..7||b5:0..1 | 00||b3:6..7||b4:0..3 | 00||b3:0..5 | 00||b2:2..7 | 00||b1:4..7||b2:0..1 | 00||b0:6..7||b1:0..3 | 00||b0:0..5 |
+            // +---------------+-------------+----------------------+----------------------+-------------+-------------+----------------------+----------------------+-------------+
+            // |   vec_0x3fs   |  00111111   |       00111111       |       00111111       |  00111111   |  00111111   |       00111111       |       00111111       |  00111111   |
+            // +---------------+-------------+----------------------+----------------------+-------------+-------------+----------------------+----------------------+-------------+
+            // | after vpextd  |   b5:0..7   |       b4:0..7        |       b3:0..7        |   b2:0..7   |   b1:0..7   |       b0:0..7        |       00000000       |  00000000   |
```

Comment:
Are theses comments correct or am I misunderstanding this? I read the final result as something starting as `b5:2..7 || b4:4..7|| b5:0..1` from vpextd.

Reply:
Because the bytes are displayed e15..e8, instead of the other way around it's hard to follow.  As an example, consider just the last four bytes of the table, but displayed in the reverse order:

```
00||b0:0..5    00||b0:6..7||b1:0..3    00||b1:4..7||b2:0..1    00||b2:2..7
```

After vpextd with bit select pattern 00111111 for all bytes:

```
b0:0..5||b0:6..7    b1:0..3||1:4..7    b2:0..1||b2:2..7
```
=
```
b0:0..7    b1:0..7    b2:0..7
```

Should I reverse the order of this table with a comment at the top, to explain the reason for the reversal?  It seems like a good idea.

Reply:
Since you are operating on doublewords here, expressing this as operations on a doubleword instead of bytes would be more intuitive here.  I think the lane mappings for little endian are what throw me off.

Reply:
Are you satisfied with the recent changes to the comments?